pandas/core/internals/base.py

```diff
@@ -98,3 +98,9 @@ def equals(self, other: object) -> bool:
             return False
 
         return self._equal_values(other)
+
+
+class SingleManager(DataManager):
```

Comment:
I initially added this class to have a common subclass for SingleBlockManager and SingleArrayManager. 

But in the end, the only way that it is actually being used are a few places that do `isinstance(mgr, SingleManager)`, while I could also replace those with `isinstance(mgr, (SingleBlockManager, SingleArrayManager))` instead of having this additional class injected in the class hierarchy.